src/pypgstac/pypgstac/load.py

```diff
@@ -1,27 +1,29 @@
 """Utilities to bulk load data into pgstac from json/ndjson."""
 import contextlib
-from datetime import datetime
 import itertools
 import logging
-from pathlib import Path
 import sys
 import time
 from dataclasses import dataclass
-from functools import lru_cache
+from datetime import datetime
+from enum import Enum
+from pathlib import Path
 from typing import (
     Any,
     BinaryIO,
     Dict,
+    Generator,
     Iterable,
     Iterator,
     Optional,
+    TextIO,
     Tuple,
     Union,
-    Generator,
-    TextIO,
 )
+
 import orjson
 import psycopg
+from methodtools import lru_cache
```

Comment:
methodtools (which I didn't know) seems pretty old and not widely used, is there any reason not to use https://cachetools.readthedocs.io/en/latest/#module-cachetools.func ?

Reply:
I didn't know there was an issue until Ruff said that the regular lru_cache has a memory issue and methodtools was the first thing googling came up with as a direct replacement. I'll change to cachetools.